### PR TITLE
[Feature-5028][MR] Support mapreduce name

### LINKED ISSUE #5028: [Feature][MR] Support mapreduce name
**Describe the feature**

Support mapreduce name, which is common option

**Is your feature request related to a problem? Please describe.**

No

### PR DESCRIPTION
**Pull request by chengshiwen**
## Purpose of the pull request

*[Feature-5028][MR] Support mapreduce name*

This closes #5028

## Brief change log

  - *Support mapreduce name*

## Verify this pull request

This pull request is already covered by existing tests, such as *(please describe tests)*.

  - *dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/utils/MapReduceArgsUtilsTest.java*

(and)

This change added tests and can be verified as follows:

  - *Manually verified the change by testing locally.*

### CODE REVIEW

**Comment by chengshiwen** [comment]
@CalvinKirs @chenxingchun Please review

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/MapReduceArgsUtils.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            args.add(String.format("%s%s=%s", Constants.D, Constants.MR_NAME, ArgsUtils.escape(appName)));
+        }
+
+        String others = param.getOthers();
+        if (StringUtils.isEmpty(others) || !others.contains(Constants.MR_QUEUE)) {
+            String queue = param.getQueue();
+            if (StringUtils.isNotEmpty(queue)) {
+                args.add(String.format("%s%s=%s", Constants.D, Constants.MR_QUEUE, queue));
+            }
+        }
```
Comment (CalvinKirs) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/MapReduceArgsUtils.java:70]: When other!=null and !others.contains(Constants.MR_QUEUE), still need to add queue

Comment (chengshiwen) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/MapReduceArgsUtils.java:70]: - `others` is `null` or `""`: `StringUtils.isEmpty(others)` is `true`, so `queue` is added
- `others` isn't `empty`: if `!others.contains(Constants.MR_QUEUE)` is `true`, `queue` is also added

The logic you mentioned is originally there

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**